### DISCUSSION TITLE
DLRS flow errors - Too many future calls: 51 and Unable to lock row

### DISCUSSION THREAD

**Discussion by TRajewich** [SFDO-Community/declarative-lookup-rollup-summaries#1510] [Q&A] [answered]
I'm not well-versed in flows and am encountering a challenge with dlrs. I've written a flow using a send email action to log to a custom object (called SHR) that is related to the Account object. The email body pulls fields from the Account. I also have a dlrs on this object that is counting the # of activities, including emails, related to this object. This is creating a mess of flow errors - Too many future calls: 51 and unable to lock row. Is there a best practice or a way around this? 

This may not be enough info. It's pretty straightforward - SHRs meeting a set criteria run through the flow when created or updated, I use a decision to determine further criteria using another related object (Opportunity), use a get records element to get the Account Id for use in the send email body, then a send email action logging to the SHR as the related record. 

Any input is greatly appreciated!

**Comment by aheber** [comment, marked as answer]
@TRajewich I think this would be a better question for the [Trailblazer Community Group](https://success.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F9300000009O5p)

DLRS doesn't create `@future` calls by itself so this is probably a symptom of your configuration at large. You'll have to put in the time to track down the various parts of the entire automation and understand why it seems to be doing a lot of work in many different parts. Likely you have something that is acting on each record when it should group them together.

**Reply by TRajewich** [reply]
Thanks. I'll try there.
